packages/flutter_tools/lib/src/ios/xcodeproj.dart

```diff
@@ -477,16 +471,34 @@ class XcodeProjectInfo {
     }
     return false;
   }
+
   /// Returns unique scheme matching [buildInfo], or null, if there is no unique
   /// best match.
-  String? schemeFor(BuildInfo? buildInfo) {
-    final String expectedScheme = expectedSchemeFor(buildInfo);
+  String? schemeFor({required String hostAppProjectName, String? flavor}) {
+    if (flavor != null) {
+      // if flavor exists, should take precedence over everything else
+      final String expectedScheme = sentenceCase(flavor);
+      if (schemes.contains(expectedScheme)) {
+        return expectedScheme;
+      }
+      return _uniqueMatch(schemes, (String candidate) {
+        return candidate.toLowerCase() == expectedScheme.toLowerCase();
+      });
+    }
+    // if theres no flavor, then regardless if the project
+    // is renamed or the default 'Runner', it should match one of the schemes
+    final String expectedScheme = sentenceCase(hostAppProjectName);
+    String? matchingScheme;
     if (schemes.contains(expectedScheme)) {
       return expectedScheme;
     }
-    return _uniqueMatch(schemes, (String candidate) {
+    matchingScheme = _uniqueMatch(schemes, (String candidate) {
```

Comment:
nit: Doesn't seem like `matchingScheme` needs to be declared before the if and looks like it can be final.
```dart
if (schemes.contains(expectedScheme)) {
  return expectedScheme;
}
final String? matchingScheme = _uniqueMatch(schemes, (String candidate) {
```